enterprise/server/test/integration/remote_cache/remote_cache_test.go

```diff
@@ -190,3 +197,93 @@ func TestBuild_RemoteCacheFlags_Compression_SecondBuildIsCached(t *testing.T) {
 	)
 }
+
+func TestBuild_RemoteCache_ScoreCard(t *testing.T) {
+	app := buildbuddy_enterprise.RunWithConfig(
+		t, buildbuddy_enterprise.NoAuthConfig,
+		"--redis_command_buffer_flush_period=0",
+		"--cache_stats_finalization_delay=0")
+	bbService := app.BuildBuddyServiceClient(t)
+	ctx := context.Background()
+	ws := testbazel.MakeTempWorkspace(t, workspaceContents)
+	iid := newUUID(t)
+	buildFlags := []string{"//:hello.txt", "--invocation_id=" + iid}
+	buildFlags = append(buildFlags, app.BESBazelFlags()...)
+	buildFlags = append(buildFlags, app.RemoteCacheBazelFlags()...)
+
+	{
+		result := testbazel.Invoke(ctx, t, ws, "build", buildFlags...)
+
+		assert.NoError(t, result.Error)
+		inv := getInvocationWithStats(t, ctx, bbService, iid)
+		sc := inv.GetScoreCard()
+		clearActionIDs(sc)
+		expectedSC := &capb.ScoreCard{
+			Misses: []*capb.ScoreCard_Result{
+				{
+					ActionMnemonic: "Genrule",
+					TargetId:       "//:hello_txt",
+				},
+			},
+		}
+		require.Equal(t, proto.MarshalTextString(expectedSC), proto.MarshalTextString(sc))
+	}
+
+	// Clear the local cache so we can try for a remote cache hit.
+	testbazel.Clean(ctx, t, ws)
+
+	{
+		iid = newUUID(t)
+		buildFlags = append(buildFlags, "--invocation_id="+iid)
+
+		result := testbazel.Invoke(ctx, t, ws, "build", buildFlags...)
+
+		assert.NoError(t, result.Error)
+		inv := getInvocationWithStats(t, ctx, bbService, iid)
+		sc := inv.GetScoreCard()
+		clearActionIDs(sc)
+		expected := &capb.ScoreCard{
+			Misses: []*capb.ScoreCard_Result{},
+		}
+		require.Equal(t, proto.MarshalTextString(expected), proto.MarshalTextString(sc))
+	}
+}
+
+func clearActionIDs(sc *capb.ScoreCard) {
+	for _, m := range sc.GetMisses() {
+		m.ActionId = ""
+	}
+}
+
+func getInvocationWithStats(t *testing.T, ctx context.Context, bbService bbspb.BuildBuddyServiceClient, iid string) *inpb.Invocation {
+	// Even though we've disabled redis buffering by setting
+	// --redis_command_buffer_flush_period=0, we still need to poll for stats to
+	// be written, since the stats recorder runs as a background job after the
+	// invocation is completed. We use a short timeout though, since the job
+	// should start immediately.
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+	r := retry.DefaultWithContext(ctx)
+	for r.Next() {
+		res, err := bbService.GetInvocation(ctx, &inpb.GetInvocationRequest{
+			Lookup: &inpb.InvocationLookup{InvocationId: iid},
+		})
+		require.NoError(t, err)
+		require.Len(t, res.Invocation, 1, "expected exactly one invocation in GetInvocationResponse")
+		inv := res.Invocation[0]
+		cs := inv.GetCacheStats()
+		if cs.GetCasCacheHits() > 0 || cs.GetCasCacheMisses() > 0 || cs.GetCasCacheUploads() > 0 {
+			return inv
+		}
+	}
+	require.FailNow(t, "Timed out waiting for invocation with cache stats")
+	return nil
+}
+
+func newUUID(t *testing.T) string {
+	id, err := uuid.NewRandom()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return id.String()
+}
```

Comment:
this could probably go in some common testutil (similar to GetTempDir)

Reply:
sg, I'll refactor in a separate PR and update everybody at once